osu.Game/Overlays/Chat/ChatTextBar.cs

```diff
@@ -0,0 +1,176 @@
+                                    Margin = new MarginPadding { Right = 2 },
+                                },
+                            },
+                            new Container
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                Padding = new MarginPadding { Right = 5 },
+                                Child = TextBox = new ChatTextBox
+                                {
+                                    Anchor = Anchor.CentreLeft,
+                                    Origin = Anchor.CentreLeft,
+                                    RelativeSizeAxes = Axes.X,
+                                    ShowSearch = { BindTarget = ShowSearch },
+                                    HoldFocus = true,
+                                    ReleaseFocusOnCommit = false,
+                                },
+                            },
+                            enterContainer = new Container
+                            {
+                                Origin = Anchor.CentreLeft,
+                                Anchor = Anchor.CentreLeft,
+                                Masking = true,
+                                BorderColour = colourProvider.Background1,
+                                BorderThickness = 2,
+                                CornerRadius = 10,
+                                Margin = new MarginPadding { Right = 10 },
+                                Size = new Vector2(60, 30),
+                                Children = new Drawable[]
+                                {
+                                    new Box
+                                    {
+                                        RelativeSizeAxes = Axes.Both,
+                                        Colour = Colour4.Transparent,
+                                    },
+                                    new OsuSpriteText
+                                    {
+                                        Text = "Enter",
+                                        Origin = Anchor.Centre,
+                                        Anchor = Anchor.Centre,
+                                        Colour = colourProvider.Background1,
+                                        Font = OsuFont.Torus.With(size: 20),
+                                        Margin = new MarginPadding { Bottom = 2 },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            ShowSearch.BindValueChanged(change =>
+            {
+                if (change.NewValue)
+                {
+                    chattingTextContainer.Hide();
+                    enterContainer.Hide();
+                    searchIconContainer.Show();
```

Comment:
This could be shortened somewhat by writing

```csharp
chattingTextContainer.FadeTo(change.NewValue ? 0 : 1);
```

etc. Also not sure on the naming of `change` - feels like it's obscuring what the change is a bit...